pkg/config/config.go

```diff
@@ -471,47 +469,55 @@ func validateGroupName(name string, usedNames map[string]bool) {
 
 // checkModuleAndGroupNames checks and imports module and resource group IDs
 // and names respectively.
-func checkModuleAndGroupNames(
-	depGroups []DeploymentGroup) (map[string]int, error) {
-	moduleToGroup := make(map[string]int)
+func checkModuleAndGroupNames(depGroups []DeploymentGroup) error {
+	seen := map[string]struct{}{}
 	groupNames := make(map[string]bool)
 	for iGrp, grp := range depGroups {
 		validateGroupName(grp.Name, groupNames)
 		for _, mod := range grp.Modules {
-			// Verify no duplicate module names
-			if _, ok := moduleToGroup[mod.ID]; ok {
-				return moduleToGroup, fmt.Errorf(
-					"%s: %s used more than once", errorMessages["duplicateID"], mod.ID)
+
+			if _, ok := seen[mod.ID]; ok {
+				return fmt.Errorf("%s: %s used more than once", errorMessages["duplicateID"], mod.ID)
 			}
-			moduleToGroup[mod.ID] = iGrp
+			seen[mod.ID] = struct{}{}
 
 			// Verify Module Kind matches group Kind
 			if grp.Kind == "" {
 				depGroups[iGrp].Kind = mod.Kind
 			} else if grp.Kind != mod.Kind {
-				return moduleToGroup, fmt.Errorf(
+				return fmt.Errorf(
 					"%s: deployment group %s, got: %s, wanted: %s",
 					errorMessages["mixedModule"],
 					grp.Name, grp.Kind, mod.Kind)
 			}
 		}
 	}
-	return moduleToGroup, nil
+	return nil
+}
+
+func modToGrp(groups []DeploymentGroup, modID string) (int, error) {
+	for iGrp, grp := range groups {
+		for _, mod := range grp.Modules {
```

Comment:
Should we consider [slices.ContainsFunc](https://pkg.go.dev/golang.org/x/exp/slices#ContainsFunc)? An example is already in pkg/config/expand.go. There is probably a marginal cost in performance, but one less thing.